gax-java/gax/src/main/java/com/google/api/gax/rpc/StubSettings.java

```diff
@@ -338,6 +352,8 @@ protected Builder(ClientContext clientContext) {
         this.quotaProjectId = getQuotaProjectIdFromClientContext(clientContext);
         this.gdchApiAudience = clientContext.getGdchApiAudience();
         this.universeDomain = clientContext.getUniverseDomain();
+        this.endpointContextBuilder =
```

Comment:
I know this is inside `StubSettings.Builder`, but instead of passing around the builder, we should pass the `EndpointContext` and call `toBuilder()` every time we need to set something, so that we won't have a incomplete `EndpointContext` passing around which could easily cause bugs later.
This means we would need to refactor other places, like setting the whole EndpointContext when we build `StubSettings`, instead of setting each field of it in setters.

Reply:
I'm passing around an incomplete Builder because `EndpointContext.build()` throws an IOException. Passing the builder avoids needing to catch an IOException inside each impacted setter (endpoint, mtls endpoint, etc).

Reply:
Create the EndpointContext in the constructor.